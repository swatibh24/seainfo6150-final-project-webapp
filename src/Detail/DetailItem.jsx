import React from "react";
import { Link } from "react-router-dom";
import styles from "./DetailItem.module.css";
import PropTypes from "prop-types";

const DetailItem = (props) => {
    //const recipe=props.recipe;
    return(
      <div className={styles.recipeContainer}>
        <div className={styles.Recipe_part}>
        <img
          className={styles.recipeImg}
          src={props.recipe.imageURL}
          alt={props.recipe.name}
        />
        <h1 className={styles.recipeName}>{props.recipe.name}</h1>
        <p className={styles.recipeDescription}>{props.recipe.description}</p>
        <h2 className={styles.recipeTitle}>Ingredients</h2>
        <p className={styles.recipeIngredients}>{props.recipe.ingredients}</p>
        <h2 className={styles.recipeTitle}>Cooking Directions</h2>
        <p className={styles.recipeDetail}>{props.recipe.one}</p>
        <p className={styles.recipeDetail}>{props.recipe.two}</p>
        <p className={styles.box}>
          <Link className={styles.catBackLink} to={props.recipe.categoryURL}>
            &lt; Back to {props.recipe.categoryName} Page
          </Link>
          <Link className={styles.recipeBackLink} to="/recipes">
        Back to All Recipes Page &gt;
      </Link>
        </p>
        </div>
      </div>
    );
    
};

DetailItem.propTypes = {
  recipe: PropTypes.object.isRequired,
};

export default DetailItem;