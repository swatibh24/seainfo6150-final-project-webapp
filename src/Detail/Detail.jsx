import React from "react";
import { Link } from "react-router-dom";
import styles from "./DetailItem.module.css";
import DetailItem from "./DetailItem.jsx";
import PropTypes from "prop-types";

const Detail = (props) => {
  let displayContent;

  if (props.recipes.length) {
    displayContent = (
        <div>
        {props.recipes.map((recipe) =>
         recipe.recipeID == props.recipeID ? 
          <DetailItem recipe={recipe} key={recipe.ID} />:null
       )}
        </div>
    );
  } else {
    displayContent = <div>You have no data!</div>;
  }

  return (
    <div >
        {displayContent}
    </div>
  );
};

Detail.propTypes = {
    recipes: PropTypes.array.isRequired,
    recipeID:PropTypes.string.isRequired
}

export default Detail;