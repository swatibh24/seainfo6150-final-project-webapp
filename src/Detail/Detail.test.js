import React from "react";
import { render } from "@testing-library/react";
import Detail from "./Detail.jsx";
import { Link, BrowserRouter } from "react-router-dom";

it("renders the Detail component correctly", () => {
  const recipes = [
    {
      "categoryID": "1",
      "recipeID": "11",
      "ID":"11-Cream-of-Mushroom-soup",
      "categoryName": "Soup",
      "name": "Cream of Mushroom Soup",
      "description": "A simple type of soup where a basic roux is thinned with cream or milk and then mushrooms and/or mushroom broth are added.",
      "ingredients": "2 tablespoons butter, 1/2 pound sliced fresh mushrooms, 1/4 cup chopped onion, 6 tablespoons all-purpose flour, 1/2 teaspoon salt, 1/8 teaspoon pepper, 2 cans (14-1/2 ounces each) chicken broth, 1 cup half-and-half cream",
      "details": {
        "one": "Step 1: In a large saucepan, heat butter over medium-high heat; saute mushrooms and onion until tender.",
        "two": "Step 2: Mix flour, salt, pepper and 1 can broth until smooth; stir into mushroom mixture. Stir in remaining broth. Bring to a boil; cook and stir until thickened, about 2 minutes. Reduce heat; stir in cream. Simmer, uncovered, until flavors are blended, about 15 minutes, stirring occasionally."
      },
      "imageURL": "https://www.budgetbytes.com/wp-content/uploads/2017/09/Creamy-Garlic-Mushroom-Soup-spoon.jpg",
      "detailURL": "/recipe/11",
      "categoryURL": "/category/1"
    },
     {
      "categoryID": "1",
      "recipeID": "12",
      "ID":"12-Mexican-Chicken-Corn-Chowder",
      "categoryName": "Soup",
      "name": "Mexican Chicken Corn Chowder",
      "description": "A creamy, cheesy, veggie and chicken filled soup that's layered with Mexican flavors.",
      "ingredients": "1-1/2 pounds boneless skinless chicken breasts, 1/2 cup chopped onion, 3 tablespoons butter, 1 to 2 garlic cloves, 1 cup hot water, 2 teaspoons chicken bouillon granules, 1/2 to 1 teaspoon ground cumin, 2 cups half-and-half cream, 2 cups shredded Monterey Jack cheese, 1 can (14-3/4 ounces) cream-style corn, 1 can (4 ounces) chopped green chiles, 1/4 to 1 teaspoon hot pepper sauce, 1 medium tomato",
      "details": {
        "one": "Step 1: In a Dutch oven, brown chicken and onion in butter until chicken is no longer pink. Add garlic; cook 1 minute longer. Add the water, bouillon and cumin; bring to a boil. Reduce heat; cover and simmer for 5 minutes.",
        "two": "Step 2: Stir in the cream, cheese, corn, chiles and hot pepper sauce. Cook and stir over low heat until cheese is melted; add tomato. If desired, top with cilantro and tortilla strips."
      },
      "imageURL": "https://myfoodstory.com/wp-content/uploads/2017/11/Instant-Pot-Chicken-Potato-Corn-Chowder-with-Bacon-Pressure-Cooker-Recipe-2.jpg",
      "detailURL": "/recipe/12",
      "categoryURL": "/category/1"
    }
  ]
  const { container } = render(<BrowserRouter><Detail recipes={recipes} recipeID="12"/></BrowserRouter>);
  expect(container).toMatchSnapshot();
});