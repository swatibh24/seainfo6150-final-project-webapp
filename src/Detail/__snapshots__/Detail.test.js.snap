// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Detail component correctly 1`] = `
<div>
  <div>
    <div>
      <div
        class="recipeContainer"
      >
        <div
          class="Recipe_part"
        >
          <img
            alt="Mexican Chicken Corn Chowder"
            class="recipeImg"
            src="https://myfoodstory.com/wp-content/uploads/2017/11/Instant-Pot-Chicken-Potato-Corn-Chowder-with-Bacon-Pressure-Cooker-Recipe-2.jpg"
          />
          <h1
            class="recipeName"
          >
            Mexican Chicken Corn Chowder
          </h1>
          <p
            class="recipeDescription"
          >
            A creamy, cheesy, veggie and chicken filled soup that's layered with Mexican flavors.
          </p>
          <h2
            class="recipeTitle"
          >
            Ingredients
          </h2>
          <p
            class="recipeIngredients"
          >
            1-1/2 pounds boneless skinless chicken breasts, 1/2 cup chopped onion, 3 tablespoons butter, 1 to 2 garlic cloves, 1 cup hot water, 2 teaspoons chicken bouillon granules, 1/2 to 1 teaspoon ground cumin, 2 cups half-and-half cream, 2 cups shredded Monterey Jack cheese, 1 can (14-3/4 ounces) cream-style corn, 1 can (4 ounces) chopped green chiles, 1/4 to 1 teaspoon hot pepper sauce, 1 medium tomato
          </p>
          <h2
            class="recipeTitle"
          >
            Cooking Directions
          </h2>
          <p
            class="recipeDetail"
          />
          <p
            class="recipeDetail"
          />
          <p
            class="box"
          >
            <a
              class="catBackLink"
              href="/category/1"
            >
              &lt; Back to 
              Soup
               Page
            </a>
            <a
              class="recipeBackLink"
              href="/recipes"
            >
              Back to All Recipes Page &gt;
            </a>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
